plugins/pipelines/app/models/concerns/samson_pipelines/stage_concern.rb

```diff
@@ -8,6 +10,34 @@ def next_stages
     Stage.find(next_stage_ids)
   end
 
+  # Needs to find all the possible stages in case this is a pipeline of pipelines as each subsequent stage
+  # could have valid next_stage_ids
+  def all_stages
```

Comment:
all stages might be a bit confusing ... how about `resursive_next_stages` or `all_next_stages` ...

also implementation can be:

```
Stage.find(recursive_next_stage_ids)
```